### PR TITLE
Cleanup redundant tests in meta after test flatten

### PR DESCRIPTION
**Pull request by rok-cesnovar**
## Summary

This PR cleans up the tests after #1476 
Also cleans up the test names to remove the scal/arr/mat distinction a bit. 

## Tests

This is all changes to tests.

## Side Effects

/
## Checklist

- [x] Math issue #937 

- [x] Copyright holder: Rok Češnovar

- [x] the basic tests are passing

- [x] the code is written in idiomatic C++ and changes are documented in the doxygen

- [x] the new changes are tested

### CODE REVIEW

**Comment by rok-cesnovar** [comment]
I now changed the base branch of this PR to develop and merged develop in. Should be good for reviewing. Its mostly renames and removal of a few redundant tests. Should be a straightforward review.

File: test/unit/math/prim/meta/contains_std_vector_test.cpp [hunk 1]
```diff
@@ -2,15 +2,16 @@
...
+TEST(MathMetaPrim, contains_vector_true) {
   using stan::contains_std_vector;
   using std::vector;
 
   EXPECT_TRUE(contains_std_vector<std::vector<double> >::value);
   EXPECT_TRUE(contains_std_vector<std::vector<int> >::value);
   EXPECT_TRUE(contains_std_vector<const std::vector<double> >::value);
   EXPECT_TRUE(contains_std_vector<const std::vector<int> >::value);
-
+  EXPECT_TRUE(contains_std_vector<std::vector<Eigen::VectorXd> >::value);
```
Comment (syclik) [inline test/unit/math/prim/meta/contains_std_vector_test.cpp]: @rok-cesnovar, thanks. Just adding a comment letting you know that I see these changes and approve them.

File: test/unit/math/prim/meta/is_eigen_test.cpp [hunk 2]
```diff
@@ -4,23 +4,7 @@
...
-  using stan::is_eigen;
-  EXPECT_FALSE((is_eigen<bool>::value));
-  EXPECT_FALSE((is_eigen<double>::value));
-  EXPECT_FALSE((is_eigen<int>::value));
-
-  EXPECT_FALSE((is_eigen<std::vector<double>>::value));
-}
-
-TEST(MathMetaPrimMat, primitive) {
+TEST(MathMetaPrim, primitive) {
```
Comment (syclik) [inline test/unit/math/prim/meta/is_eigen_test.cpp]: Same here. I see these changes and know why they're happening.

**Comment by rok-cesnovar** [comment]
Thanks for the review. Will move to the actual meta flatten now.

**Pull request merged**